### PR TITLE
PAAS-1282: Add HTTP Security Headers to Haproxy  on update

### PR DESCRIPTION
**Pull request by amitc-jahia**
JIRA issue: https://jira.jahia.org/browse/PAAS-1282

Short description:
Add HTTP Security Response Headers at HAProxy level on update

### CODE REVIEW

File: jahia-cloud-env-update/update-to-3.2.0/update.yml [hunk 1]
```diff
@@ -60,3 +77,15 @@ actions:
...
+  addHTTPSecuityHeaders:
+    cmd[bl]: |-
+      cp /etc/haproxy/haproxy.cfg.d/00-global.cfg /etc/haproxy/haproxy.cfg.d/00-global.cfg.bak
+      insert_line_1='    http-response set-header Strict-Transport-Security "max-age=63072000;"'
+      insert_line_2='    http-response set-header X-Content-Type-Options "nosniff"'
+      insert_line_3='    http-response set-header X-XSS-Protection "1; mode=block;"'
+      pattern='(Host) len 50'
+      sed -ie "/$pattern/a\\\n$insert_line_1\n$insert_line_2\n$insert_line_3" /etc/haproxy/haproxy.cfg.d/00-global.cfg
+      chown -R haproxy: /etc/haproxy
+      service haproxy reload
```
Comment (laurentfufu) [inline jahia-cloud-env-update/update-to-3.2.0/update.yml:90]:
```suggestion
      insert_line_1='    http-response set-header Strict-Transport-Security "max-age=63072000;"'
      insert_line_2='    http-response set-header X-Content-Type-Options "nosniff"'
      insert_line_3='    http-response set-header X-XSS-Protection "1; mode=block;"'
      if ! (grep -qPzo "$insert_line_1\n$insert_line_2\n$insert_line_3" /etc/haproxy/haproxy.cfg.d/00-global.cfg); then
        echo "headers already there"
        exit 0
      fi
      pattern='(Host) len 50'
      sed -i.bak "/$pattern/a\\\n$insert_line_1\n$insert_line_2\n$insert_line_3" /etc/haproxy/haproxy.cfg.d/00-global.cfg
      chown haproxy: /etc/haproxy/00-global.cfg.bak
      service haproxy reload
```
not sure about the `.bak` is relevant here but why not.

the added `grep` is not tested but should be use for  not having several time the headers line if manifest have to be applied several time on an env.

**Review by laurentfufu — CHANGES_REQUESTED** [review]

File: jahia-cloud-env-update/update-to-3.2.0/update.yml [hunk 2]
```diff
@@ -60,3 +77,15 @@ actions:
...
+    user: root
```
Comment (kevinvandel) [inline jahia-cloud-env-update/update-to-3.2.0/update.yml:91]: No newline at end of file :scream:

File: jahia-cloud-env-update/update-to-3.2.0/update.yml [hunk 3]
```diff
@@ -60,3 +77,15 @@ actions:
...
       fi
     user: root
+    
+  addHTTPSecuityHeaders:
+    cmd[bl]: |-
+      cp /etc/haproxy/haproxy.cfg.d/00-global.cfg /etc/haproxy/haproxy.cfg.d/00-global.cfg.bak
+      insert_line_1='    http-response set-header Strict-Transport-Security "max-age=63072000;"'
+      insert_line_2='    http-response set-header X-Content-Type-Options "nosniff"'
+      insert_line_3='    http-response set-header X-XSS-Protection "1; mode=block;"'
+      pattern='(Host) len 50'
```
Comment (Ronan-Gorain) [inline jahia-cloud-env-update/update-to-3.2.0/update.yml:87]: This pattern could be used in customer rewrite rules which are added in the same file (but not visible here).

**Comment by amitc-jahia** [comment]
Requested Changes have been made. Please Review.

**Review by laurentfufu — CHANGES_REQUESTED** [review]
- with this pattern, resulting stanza will not have the same order as https://github.com/Jahia/paas_jelastic_dx_universal/pull/150/files which should potentially be a problem for future sed update if the haproxy nodes haven't been redeployed since

- still not sure about the backup file benefit because it spoil (in a way) the haproxy's conf folder, maybe keeping it in the root home user with the update version in the suffix in addition to the .bkp one, but basically  IMO, we don't want to rollback the added line so we don't need this backup file

**Comment by amitc-jahia** [comment]
Following changes have been made. Please review.

- The pattern has been changed to maintain the same order as https://github.com/Jahia/paas_jelastic_dx_universal/pull/150/files and also keeping in mind with the customer `REWRITES`.

- Based on the suggestion of not rolling back to this backup file, the backup file has been removed.

**Review by laurentfufu — APPROVED** [review]

**Review by laurentfufu — APPROVED** [review]

**Pull request merged**